aten/src/ATen/native/cuda/Loss.cu

```diff
@@ -61,12 +61,14 @@ void binary_cross_entropy_backward_out_kernel(Tensor& grad_input, const Tensor&
 namespace at { namespace native {
 
 Tensor kl_div_backward_cuda(const Tensor& grad, const Tensor& input, const Tensor& target, int64_t reduction, bool log_target) {
-  auto grad_input = at::empty_like(input);
+  const auto grad_expanded = grad.expand_as(input);
```

Comment:
Hmm backward wasn't failing previously right?

Reply:
Backward is all good, but this change breaks JIT for some reason.

Reply:
No, it was not because the tests used `target.numel() == input.numel()` inputs.

Reply:
Oh, I was thinking that because grad has the same size as the potentially reduced output, we'd need to do some kind of expanding always, and whether target was broadcasted to input or not doesn't matter. But maybe TensorIterator is doing something?

Reply:
TI was complaining about shape mismatch in some cases, and in some it would segfault. Probably worth investigating...

Reply:
Yeah tensor iterator is supposed to handle broadcasting, but maybe there is an issue specifically with this case?

Reply:
Either that, or the grad forwarding. I will try to isolate the issue, maybe it is something worth reporting about.

Reply:
My understanding is that, before the fix, TI didn't know to broadcast because it was handed an unbroadcasted `target` and `grad`.

Reply:
TI should have been broadcasted inputs with shape (5, 5, 5) and (5,) (where it complaints about output-input mismatch), and shapes (5, 5, 5), (5, 5) should not have caused any segfaults. Back then TI used to broadcast inputs and outputs to the same shape, but, apparently, this behavior might have been changed, CC @ngimel .